### PR TITLE
Adds Collection.flatMapBatches(of:).

### PR DESCRIPTION
**Pull request by jasdev**
h/t @freak4pc and @natecook1000 for this implementation from a thread in iOS Folks. 👏🏽

### CODE REVIEW

**Comment by freak4pc** [comment]
Thanks for putting this together! I'm having a hard time with the name though :) Got any other ideas?

**Comment by jasdev** [comment]
How's `Collection.batchedPublisher(by:)` to better align with the `.publisher` property convention?

**Comment by freak4pc** [comment]
What about `flatMapBatches(of size: Int)` ?

**Comment by jasdev** [comment]
@freak4pc, done and done. ✅

**Review by freak4pc — APPROVED** [review]
Love it. Thank you so much!

File: Sources/Operators/FlatMapBatches.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    func flatMapBatches(of size: Int) -> AnyPublisher<[Element.Output], Element.Failure> {
+        precondition(size > 0, "Batch sizes must be positive.")
+
+        let indexBreaks = sequence(
+            first: startIndex,
+            next: {
+                $0 == endIndex ?
+                    nil :
+                    index($0, offsetBy: size, limitedBy: endIndex)
+                        ?? endIndex
```
Comment (freak4pc) [inline Sources/Operators/FlatMapBatches.swift:33]: This is the Apple foundation team voodoo alright! 😂

**Pull request merged**